File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLUserStatusLastWeek.java
package com.github.badoualy.telegram.tl.api;

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public class TLUserStatusLastWeek extends TLAbsUserStatus {
    public static final int CONSTRUCTOR_ID = 0x7bf09fc;

    public TLUserStatusLastWeek() {
    }

    @Override
    public String toString() {
        return "userStatusLastWeek#7bf09fc";
    }

    @Override
    public int getConstructorId() {
        return CONSTRUCTOR_ID;
    }

    @Override
    @SuppressWarnings("PointlessBooleanExpression")
    public boolean equals(Object object) {
        if (!(object instanceof TLUserStatusLastWeek)) return false;
        if (object == this) return true;

        TLUserStatusLastWeek o = (TLUserStatusLastWeek) object;

        return true;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLUserStatusLastWeek.kt
package com.github.badoualy.telegram.tl.api

/**
 * userStatusLastWeek#7bf09fc
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLUserStatusLastWeek : TLAbsUserStatus() {
    private val _constructor: String = "userStatusLastWeek#7bf09fc"

    override val constructorId: Int = CONSTRUCTOR_ID

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLUserStatusLastWeek) return false
        if (other === this) return true

        return true
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0x7bf09fc.toInt()
    }
}
